packages/web-components/fast-foundation/src/badge/badge.template.ts

```diff
@@ -1,14 +1,19 @@
 import { ElementViewTemplate, html } from "@microsoft/fast-element";
-import type { FASTBadge } from "./badge.js";
+import { endSlotTemplate, startSlotTemplate } from "../patterns/index.js";
+import type { BadgeOptions, FASTBadge } from "./badge.js";
 
 /**
  * The template for the {@link @microsoft/fast-foundation#FASTBadge} component.
  * @public
  */
-export function badgeTemplate(): ElementViewTemplate<FASTBadge> {
+export function badgeTemplate(
+    options: BadgeOptions = {}
+): ElementViewTemplate<FASTBadge> {
     return html<FASTBadge>`
-        <div class="control" part="control">
+        ${startSlotTemplate(options)}
+        <div class="content" part="content">
```

Comment:
I've been wondering if we need these elements here and on button, anchor...my assumption is that the intent for it would be scenarios where truncation perhaps need to take place, so I do think it probably serves a functional purpose. I do think we'd prefer to go with a `span` rather than a `div` here and in anchor/button as opposed to div. Both are essentially structural elements but span is inline rather than block by default which I think is a better fit.

Reply:
Good call on `span`. Yes, I think it's exactly to support truncation, or really any styling that needs to target the content separate from the slots. I think color has come up before as well.

Reply:
From a semantic standpoint, the `display` isn't the only thing that differs between a `span` and a `div`.

A `div` allows for [flow content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#flow_content), and `span` allows for [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#phrasing_content). This changes what elements are allowed as children, which can matter in some cases. For instance, any `a` element is allowed in a `div`. For `span`, `a` elements are only allowed if they only contain phrasing content.

Reply:
yeah, even with this caveat I think we for sure want a span here rather than div as the semantics of badge really lend themselves to phrasing content and it's considerations.